src/logicblocks/event/processing/broker/broker/broker_builder.py

```diff
@@ -0,0 +1,124 @@
+from abc import ABC, abstractmethod
+from datetime import timedelta
+from typing import Self, TypedDict
+
+from ..coordinator import EventSubscriptionCoordinator
+from ..locks import LockManager
+from ..nodes import NodeManager, NodeStateStore
+from ..observer import EventSubscriptionObserver
+from ..sources import (
+    EventStoreEventSourceFactory,
+    InMemoryEventSubscriptionSourceMappingStore,
+)
+from ..subscribers import (
+    EventSubscriberManager,
+    EventSubscriberStateStore,
+    InMemoryEventSubscriberStore,
+)
+from ..subscriptions import EventSubscriptionStateStore
+from .broker import CoordinatorObserverEventBroker, EventBroker
+
+
+class PrepareResults(TypedDict):
+    node_state_store: NodeStateStore
+    event_subscriber_state_store: EventSubscriberStateStore
+    event_subscription_state_store: EventSubscriptionStateStore
+    lock_manager: LockManager
+    event_source_factory: EventStoreEventSourceFactory
+
+
+class EventBrokerBuilder[**P = ...](ABC):
+    node_id: str
+
+    node_state_store: NodeStateStore
+    event_subscriber_state_store: EventSubscriberStateStore
+    event_subscription_state_store: EventSubscriptionStateStore
+    lock_manager: LockManager
+    event_source_factory: EventStoreEventSourceFactory
+
+    def __init__(self, node_id: str):
+        self.node_id = node_id
+
+    @abstractmethod
+    def _prepare(self, *args: P.args, **kwargs: P.kwargs) -> PrepareResults:
+        pass
+
+    def prepare(self, *args: P.args, **kwargs: P.kwargs) -> Self:
+        prepare = self._prepare(*args, **kwargs)
+        self.node_state_store = prepare["node_state_store"]
+        self.event_subscriber_state_store = prepare[
+            "event_subscriber_state_store"
+        ]
+        self.event_subscription_state_store = prepare[
+            "event_subscription_state_store"
+        ]
+        self.lock_manager = prepare["lock_manager"]
+        self.event_source_factory = prepare["event_source_factory"]
+
+        return self
+
+    def build(
+        self,
+        node_manager_heartbeat_interval: timedelta = timedelta(seconds=10),
+        node_manager_purge_interval: timedelta = timedelta(minutes=1),
+        node_manager_node_max_age: timedelta = timedelta(minutes=10),
+        subscriber_manager_heartbeat_interval: timedelta = timedelta(
+            seconds=10
+        ),
+        subscriber_manager_purge_interval: timedelta = timedelta(minutes=1),
+        subscriber_manager_subscriber_max_age: timedelta = timedelta(
+            minutes=10
+        ),
+        coordinator_subscriber_max_time_since_last_seen: timedelta = timedelta(
+            seconds=60
+        ),
+        coordinator_distribution_interval: timedelta = timedelta(seconds=20),
+        observer_synchronisation_interval: timedelta = timedelta(seconds=20),
```

Comment:
I'd quite like to wrap these up into an `EventBrokerSettings` object.

Reply:
that would be cute